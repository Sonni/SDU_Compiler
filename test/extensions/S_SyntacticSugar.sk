# Since most of the syntatic sugar is just a translation to something else
# (for loops -> while loops) (i++ -> i = i + 1) etc. Then it is only
# necessary to check if the translation is correct. This can be done by visual
# inspection of the pretty printer and output.

var rec: record of {a: int, b: bool};

allocate rec;

for rec.a = 1; rec.a < 10; rec.a += 3; do
{
    write rec.a;
}

for rec.a = 3; to 5 do
{
    write rec.a;
}

rec.a++;
rec.a--;
rec.a /= 1;
rec.a *= 2;
rec.a += 10;
rec.a -= 10;
write rec.a;    # 10

rec.a = rec.a > 100 ? 30 : 40;

write rec.a;    # 40


rec = {30, true};

write rec.a;    # 30

rec.a = -4;

write rec.a;
