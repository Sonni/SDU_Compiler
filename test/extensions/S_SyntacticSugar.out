1
4
7
3
4
10
40
30
-4
